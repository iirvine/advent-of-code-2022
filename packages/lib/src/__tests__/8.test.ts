import {
  Bounds,
  Coord,
  day8_pt1,
  day8_pt2,
  INPUT,
  iterateWithinBounds,
  parse,
  traverseFrom,
} from "../8-treehouse";

const input = `
30373
25512
65332
33549
35390`.trim();

const output = parse(input);

test("iterateWithinBounds", () => {
  const b: Bounds = [output[1][1].coord, output.at(-2)?.at(-2)?.coord as Coord];
  console.log(b);
  const result = [...iterateWithinBounds(output, b)];
  expect(result).toMatchSnapshot();
});

test("transverseFrom", () => {
  let result = [...traverseFrom(output, [0, 0], "right")];
  console.log(result);

  result = [...traverseFrom(output, [0, 0], "down")];
  console.log(result);

  result = [...traverseFrom(output, [1, 1], "right")];
  console.log(result);
});

test("day8_pt1", () => {
  const ans = day8_pt1(INPUT);
  console.log(ans);
});

test("day8_pt2", () => {
  const ans = day8_pt2(INPUT);
  console.log(ans);
});
