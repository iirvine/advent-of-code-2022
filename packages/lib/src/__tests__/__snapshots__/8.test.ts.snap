// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iterateWithinBounds 1`] = `
[
  {
    "coord": [
      1,
      1,
    ],
    "value": 5,
  },
  {
    "coord": [
      2,
      1,
    ],
    "value": 5,
  },
  {
    "coord": [
      3,
      1,
    ],
    "value": 1,
  },
  {
    "coord": [
      1,
      2,
    ],
    "value": 5,
  },
  {
    "coord": [
      2,
      2,
    ],
    "value": 3,
  },
  {
    "coord": [
      3,
      2,
    ],
    "value": 3,
  },
  {
    "coord": [
      1,
      3,
    ],
    "value": 3,
  },
  {
    "coord": [
      2,
      3,
    ],
    "value": 5,
  },
  {
    "coord": [
      3,
      3,
    ],
    "value": 4,
  },
]
`;
